ArduPlane/ArduPlane.cpp

```diff
@@ -703,4 +706,27 @@ void Plane::publish_osd_info()
 }
 #endif
 
+#if PRECISION_LANDING == ENABLED
+void Plane::init_precland()
```

Comment:
System.cpp line 142

Reply:
That's calling precland init directly, the helper function declared here isn't used though. (Calling it directly is fine, I'd just remove the unused helper then)

Reply:
> That's calling precland init directly, the helper function declared here isn't used though. (Calling it directly is fine, I'd just remove the unused helper then)

I am very sorry, you are completely right. I just bypassed the helper completely. I've already corrected it, calling the helper instead, as it is done in copter.

I've also separated the precland declarations in precision_landing.cpp as it is done in copter ( I'd swear it wasn't in a separate file like that back when I did it ) And I have put the condition for AP_TERRAIN_AVAILABLE right there, in precision_landing.cpp. This last condition is a bit ugly placed in there, but if placed only in the condition affecting it I needed to split the if-else in two ifs and thus it was uglier I think because the first condition would always evaluate.

I also added an additional commit for logging support for precland, it wasn't uploaded to my remote repository and I forgot to add it when I made this PR, sorry.

I've rebased everything to master again and I've squashed the commits regarding your suggestions. Let me know if you see anything else that should be corrected.

Thank you!